sql/plan/showcolumns.go

```diff
@@ -187,6 +187,12 @@ func (s ShowColumns) WithChildren(children ...sql.Node) (sql.Node, error) {
 	return &s, nil
 }
 
+// CheckPrivileges implements the interface sql.Node.
+func (s *ShowColumns) CheckPrivileges(ctx *sql.Context, opChecker sql.PrivilegedOperationChecker) bool {
+	// The table won't be visible during the resolution step if the user doesn't have the correct privileges
```

Comment:
Again, tests

It also seems strange that we wouldn't just call `Child.CheckPrivileges`, we have the table in the node